### ISSUE TITLE
hare: Clean up and/or prioritize all old Hare-related issues

### ISSUE THREAD

**Issue by jonZlotnik** [spacemeshos/go-spacemesh#3286]
### Ready for any Dev

- [x] #3138 
- [x] #1143 
Flaky tests:
- [x] #3079 
- [x] #2953 
- [x] #2923 
### Ready for dev with good understanding of hare
- [x] #673
- [x] #3287 
- [x] #1467 
- [x] #1331 
### Blocked by ambiguities
i.e. Somebody who knows about the issue's context needs add details.
- [x] #2544 
Genesis flow:
- [x] #953 
- [x] #1299 
- [x] #1311 
### Requires analysis with sandwich team
- [x] #767 
- [ ] #2785 
- [x] #1652 
- [x] #1968 
- [x] #1965 

### Lower priority
Code quality things
- [x] #3335 
- [x] #3334 
Probably not relevant anymore:
- [x] #1259 
- [x] #1178
